frontend/projects/ui/src/app/util/get-marketplace.ts

```diff
@@ -0,0 +1,9 @@
+import { first } from 'rxjs/operators'
+import { PatchDbService } from 'src/app/services/patch-db/patch-db.service'
+import { UIMarketplaceData } from 'src/app/services/patch-db/data-model'
+
+export function getMarketplace(
+  patch: PatchDbService,
+): Promise<UIMarketplaceData> {
+  return patch.watch$('ui', 'marketplace').pipe(first()).toPromise()
```

Comment:
@MattDHill and I discussed changing the approach here - rather than using promises to fit in with the current schema, what do you think about having each component subscribing to this data, and using `async` in the html or using the `DestroyService` in the `<component>.ts` if we need to apply logic to the data. Matt can elaborate more on this idea.